pages/library/_viewId.vue

```diff
@@ -157,6 +198,127 @@ export default Vue.extend({
     }
   },
   methods: {
+    async filterMedia() {
```

Comment:
This mostly looks like stuff that could be done inside of the filter menu itself.

It would generate the filter string internally and pass it through the `change` event.

This would have the benefit of avoiding the need to redo this entire method on every page that wants to use the filter menu.